test/extra/RpcClientServerTest.cpp

```diff
@@ -10,28 +10,119 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include <gtest/gtest.h>
+#include <up-cpp/communication/RpcClient.h>
+#include <up-cpp/communication/RpcServer.h>
+#include <up-cpp/datamodel/builder/Payload.h>
+#include <up-cpp/datamodel/builder/Uuid.h>
+#include <up-cpp/datamodel/serializer/UUri.h>
+#include <up-cpp/datamodel/serializer/Uuid.h>
+#include <up-transport-zenoh-cpp/ZenohUTransport.h>
+
+#include <iostream>
+
+using namespace std::chrono_literals;
 
 namespace {
 
-class TestFixture : public testing::Test {
+using namespace uprotocol::v1;
+using uprotocol::communication::RpcClient;
+using uprotocol::communication::RpcServer;
+
+constexpr std::string_view ZENOH_CONFIG_FILE = BUILD_REALPATH_ZENOH_CONF;
+
+struct MyUUri {
+	std::string auth = "";
+	uint32_t ue_id = 0x8000;
+	uint32_t ue_version_major = 1;
+	uint32_t resource_id = 1;
+
+	operator uprotocol::v1::UUri() const {
+		UUri ret;
+		ret.set_authority_name(auth);
+		ret.set_ue_id(ue_id);
+		ret.set_ue_version_major(ue_version_major);
+		ret.set_resource_id(resource_id);
+		return ret;
+	}
+
+	std::string to_string() const {
+		return std::string("<< ") + UUri(*this).ShortDebugString() + " >>";
+	}
+};
+
+class RpcClientServerTest : public testing::Test {
 protected:
+	MyUUri rpc_service_uuri_{"me_authority", 65538, 1, 32600};
+	MyUUri ident_{"me_authority", 65538, 1, 0};
```

Comment:
Nitpick (sorta) - Can you change the ue_id and resource_id to hexadecimal please.  I think its easier to read that way and easier to see high/low bits that way.